src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService {
+    @Override
+    public List<? super Machine> getAll(Class type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        }
+        if (type == Truck.class) {
+            return new TruckProducer().get();
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void startWorking(List list) {
+
```

Comment:
add implementation forEach machine: doWork()

Reply:
<img width="795" alt="image" src="https://user-images.githubusercontent.com/81781479/217244052-d2870373-3b44-4ca2-aa55-f3375f31c376.png">
not fixed